ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/EpochProcessingUtilTest.java

```diff
@@ -0,0 +1,236 @@
+ */
+
+package tech.pegasys.artemis.statetransition.util;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.google.common.primitives.UnsignedLong;
+import java.util.ArrayList;
+import java.util.List;
+import net.consensys.cava.bytes.Bytes32;
+import net.consensys.cava.bytes.Bytes48;
+import net.consensys.cava.junit.BouncyCastleExtension;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import tech.pegasys.artemis.datastructures.Constants;
+import tech.pegasys.artemis.datastructures.blocks.Eth1Data;
+import tech.pegasys.artemis.datastructures.operations.Deposit;
+import tech.pegasys.artemis.datastructures.operations.DepositData;
+import tech.pegasys.artemis.datastructures.operations.DepositInput;
+import tech.pegasys.artemis.datastructures.state.Validator;
+import tech.pegasys.artemis.datastructures.state.Validators;
+import tech.pegasys.artemis.statetransition.BeaconState;
+
+@ExtendWith(BouncyCastleExtension.class)
+class EpochProcessingUtilTest {
+
+  BeaconState createArbitraryBeaconState(int numValidators) {
+    ArrayList<Deposit> deposits = new ArrayList<Deposit>();
+    int count = 0;
+    // create initial validators
+    while (count < numValidators) {
+      List<Bytes32> merkle_roots = new ArrayList<Bytes32>();
+      merkle_roots.add(Bytes32.random());
+      List<Bytes48> proof_of_posession = new ArrayList<Bytes48>();
+      proof_of_posession.add(Bytes48.random());
+      proof_of_posession.add(Bytes48.random());
+      DepositInput deposit_input =
+          new DepositInput(Bytes48.random(), Bytes32.random(), proof_of_posession);
+      DepositData deposit_data =
+          new DepositData(deposit_input, UnsignedLong.valueOf(32000000000L), UnsignedLong.ZERO);
+      Deposit deposit = new Deposit(merkle_roots, UnsignedLong.ZERO, deposit_data);
+      deposits.add(deposit);
+      count++;
+    }
+    // get initial state
+    BeaconState state =
+        BeaconStateUtil.get_initial_beacon_state(
+            deposits,
+            UnsignedLong.valueOf(Constants.GENESIS_SLOT),
+            new Eth1Data(Bytes32.ZERO, Bytes32.ZERO));
+
+    UnsignedLong currentEpoch = BeaconStateUtil.get_current_epoch(state);
+
+    // set validators to active
+    for (Validator validator : state.getValidator_registry()) {
+      validator.setActivation_epoch(currentEpoch);
+    }
+    return state;
+  }
+
+  @Test
+  void processEjectionsTest() {
+
+    BeaconState state = createArbitraryBeaconState(25);
+    UnsignedLong currentEpoch = BeaconStateUtil.get_current_epoch(state);
+
+    Validators before_active_validators =
+        ValidatorsUtil.get_active_validators(state.getValidator_registry(), currentEpoch);
+    UnsignedLong before_total_balance =
+        EpochProcessorUtil.get_total_effective_balance(state, before_active_validators);
+
+    // make 4 validators have balance below threshold and 1 right at the threshhold
```

Comment:
Might be good to use the `EJECTION_BALANCE` constant here for transparency, and also to avoid breaking the tests if the value gets changed at some point. E.g. `0L`, `EJECTION_BALANCE / 2`, `EJECTION_BALANCE - 1`, `EJECTION_BALANCE`.

Reply:
great idea